src/couchConfig.js

```diff
@@ -0,0 +1,391 @@
+/*
+Copyright 2017-2018 OCAD University
+Licensed under the New BSD license. You may not use this file except in
+compliance with one these Licenses. You may obtain a copy of the BSD License at
+https://raw.githubusercontent.com/fluid-project/fluid-couch-config/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion");
+require("./retrying");
+
+var isEqual = require("underscore").isEqual;
+
+fluid.defaults("fluid.couchConfig", {
+    gradeNames: "fluid.component",
+    couchOptions: {
+        couchUrl: "http://localhost:5984",
+        dbName: null // To be provided
+    },
+    dbDocuments: {
+        // An object whose keys are the IDs of documents to
```

Comment:
As noted in https://issues.fluidproject.org/browse/SJRK-318?focusedCommentId=40849 this usage is incompatible with the immutable options introduced in the FLUID-6145 branch. Given these documents don't seem to be read from anywhere it might be more appropriate to resolve them from returned promise values rather than mutating them in place in the arguments list.